systems/controllers/QPCommon.cpp

```diff
@@ -356,16 +356,22 @@ int setupAndSolveQP(NewQPControllerData *pdata, std::shared_ptr<drake::lcmt_qp_c
   Vector6d body_vdot;
 
   for (int i=0; i < qp_input->num_tracked_bodies; i++) {
-    int body_id0 = qp_input->body_motion_data[i].body_id - 1;
-    double weight = params->body_motion[body_id0].weight;
-    desired_body_accelerations[i].body_id0 = body_id0;
+    if (qp_input->body_motion_data[i].body_id == 0)
+      mexErrMsgTxt("Body motion data with body id 0\n");
+    int body_or_frame_id0 = qp_input->body_motion_data[i].body_id - 1;
+    int true_body_id0 = pdata->r->parseBodyOrFrameID(body_or_frame_id0, NULL);
```

Comment:
In the reactive recovery work we wanted to be able to make body motion constraints on frames within the feet; this makes sure that if it was a frame ID given to the body motion data struct, the appropriate bodyid is fetched in its place (the TF to the frame within the body is extracted at another point).